src/con_duct/__main__.py

```diff
@@ -96,6 +96,21 @@ def max(self, other: ProcessStats) -> ProcessStats:
             timestamp=max(self.timestamp, other.timestamp),
         )
 
+    def __post_init__(self) -> None:
+        self._validate()
+
+    def _validate(self) -> None:
+        if not isinstance(self.pcpu, (float, int)):
+            raise TypeError(f"Expected 'pcpu' to be of type 'float' or 'int', got {type(self.pcpu).__name__}")
+        if not isinstance(self.pmem, (float, int)):
+            raise TypeError(f"Expected 'pmem' to be of type 'float' or 'int', got {type(self.pmem).__name__}")
+        if not isinstance(self.rss, int):
+            raise TypeError(f"Expected 'rss' to be of type 'int', got {type(self.rss).__name__}")
+        if not isinstance(self.vsz, int):
+            raise TypeError(f"Expected 'vsz' to be of type 'int', got {type(self.vsz).__name__}")
+        if not isinstance(self.timestamp, str):
+            raise TypeError(f"Expected 'timestamp' to be of type 'str', got {type(self.timestamp).__name__}")
```

Comment:
I really dislike duplication. Would be better to have something like
```suggestion
        for f, types in (('pcpu', (float, int)), ...):
            value = getattr(self, f)
            if not isinstance(value, types):
                raise TypeError(f"Expected '{f}' to be of type {' or '.join(map(repr(types)))}, got {type(value).__name__}")
```

but also wondered if there is some runtime way to take those from type annotations to avoid duplication here or just do "runtime type validation" (like with pydantic)?

Reply:
Considering this approach makes me wonder if we are going in the wrong direction.

Instead of blowing up if we get an invalid value, it would probably be more helpful to just record NoneType. (I'm thinking of pandas NaN as an example)

In the case where we get invalid information, we should just mark it as a broken sample, log the error, but continue with the execution

Reply:
we could. your call

Reply:
It would be pretty obnoxious to deal with the possibility of other types all over the code to cover a hypothetical case, so for now I went with the simplest runtime type validation I could do. We can come back to it if we get weird data from `ps`